command-line/lib/Command/Address/Payment.hs

```diff
@@ -65,28 +59,48 @@ mod liftCmd = command "payment" $
             , indent 2 $ string ""
             , indent 2 $ bold $ string "$ cat addr.prv \\"
             , indent 4 $ bold $ string $ "| "<>progName<>" key public --with-chain-code \\"
-            , indent 4 $ bold $ string $ "| "<>progName<>" address payment --from-key --network-tag testnet"
+            , indent 4 $ bold $ string $ "| "<>progName<>" address payment --network-tag testnet"
             , indent 2 $ string "addr_test1vqrlltfahghjxl5sy5h5mvfrrlt6me5fqphhwjqvj5jd88cccqcek"
             ])
   where
     parser = Cmd
-        <$> credentialOpt
-        <*> networkTagOpt Shelley
+        <$> networkTagOpt Shelley
 
 run :: Cmd -> IO ()
-run Cmd{networkTag,credentialType} = do
-    case (mkNetworkDiscriminant . fromIntegral . unNetworkTag) networkTag of
-        Left ErrWrongNetworkTag{} -> do
-            fail "Invalid network tag. Must be between [0, 15]"
-        Right discriminant -> do
-            addr <- case credentialType of
-                CredentialFromKey -> do
-                    xpub <- hGetXPub stdin
-                    pure $ Shelley.paymentAddress discriminant (PaymentFromKey $ Shelley.liftXPub xpub)
-                CredentialFromScript -> do
-                    scriptHash <- hGetScriptHash stdin
-                    pure $ Shelley.paymentAddress discriminant (PaymentFromScript scriptHash)
-            B8.hPutStr stdout $ T.encodeUtf8 $ bech32With hrp addr
+run Cmd{networkTag} = do
+    discriminant <- fromNetworkTag networkTag
+    (hrp, bytes) <- hGetBech32 stdin allowedPrefixes
+    addr <- addressFromBytes discriminant bytes hrp
+    hPutBytes stdout (unAddress addr) (EBech32 addrHrp)
   where
-    hrp | networkTag == shelleyTestnet = [humanReadablePart|addr_test|]
-        | otherwise = [humanReadablePart|addr|]
+    addrHrp
+        | networkTag == shelleyTestnet = CIP5.addr_test
+        | otherwise = CIP5.addr
+
+    -- TODO: Also allow `XXX_vk` prefixes. We don't need the chain code to
+    -- construct a payment credential. This will however need some additional
+    -- abstraction over `xpubFromBytes` but I've done enough yake-shaving at
```

Comment:
we can use https://github.com/input-output-hk/cardano-addresses/blob/master/core/lib/Cardano/Address/Derivation.hs#L135 and then https://github.com/input-output-hk/cardano-addresses/blob/master/core/lib/Cardano/Address/Derivation.hs#L147 right?

Reply:
We cannot actually :/
Because xpubFromBytes expects 64 bytes of inputs, both a public key and the chain code. However, here, we would only have the public key.

We could use a "dummy" chain code filled with 0 bit, but that's quite hacky/ugly.

Since that wasn't supported before this PR anyway, I didn't really do anything to add it. But I think we should later.